.githooks/pre-commit

```diff
@@ -6,3 +6,10 @@ ensure_rustup_installed
 ensure_rustfmt_installed
 
 check_formatting
+
+cd $(git rev-parse --show-toplevel)
+for i in `git ls-tree -d -r --name-only HEAD`; do
+	[ -e $i.rs ] && hook_failure "please rename $i.rs to $i/mod.rs." && exit 1
+done
+
+exit 0
```

Comment:
There is a clippy warning for this. I do not know why it didn't trigger in your case but recently I tried to create a `foo/foo.rs` at a work project and it complained.

In any case, this is something rust tooling should handle. Please remove this.

Reply:
> I do not know why it didn't trigger

Oh right, you had added an exception in the code. :facepalm:

Reply:
I don't follow "you had added an exception". no my clippy doesn't warn, but we can drop the last commit

Reply:
> I don't follow "you had added an exception".

The `#[allow(clippy::module_inception)]` that I pointed out [not too long ago](https://github.com/dbus2/zbus/pull/989#pullrequestreview-2338445323). :)

Reply:
it's not the same check, this one is for what you asked me last wrt foo/mod.rs.